File: src/main/java/com/dh/ondot/schedule/infra/exception/OdsayTooCloseException.java
package com.dh.ondot.schedule.infra.exception;

import com.dh.ondot.core.exception.BadRequestException;

import static com.dh.ondot.core.exception.ErrorCode.ODSAY_TOO_CLOSE;

public class OdsayTooCloseException extends BadRequestException {
  public OdsayTooCloseException(String detail) {
    super(ODSAY_TOO_CLOSE.getMessage().formatted(detail));
  }
  @Override
  public String getErrorCode() {
    return ODSAY_TOO_CLOSE.name();
  }
}

File: src/main/kotlin/com/dh/ondot/schedule/infra/exception/OdsayTooCloseException.kt
package com.dh.ondot.schedule.infra.exception

import com.dh.ondot.core.exception.BadRequestException
import com.dh.ondot.core.exception.ErrorCode.ODSAY_TOO_CLOSE

class OdsayTooCloseException(detail: String) :
    BadRequestException(ODSAY_TOO_CLOSE.message.format(detail)) {
    override val errorCode: String get() = ODSAY_TOO_CLOSE.name
}
